### DISCUSSION TITLE
Reliably hiding . and ..

### DISCUSSION THREAD

**Discussion by mikezackles** [alexluigit/dirvish#209] [Q&A]
I've tried using variations on this code:

```lisp
(use-package dired-x
  ;; Always launch dired in omit mode
  :hook (dired-mode . dired-omit-mode)
  :config
  ;; Hide . and ..
  (setq dired-omit-files "^\\.\\.?$")) ; Note that I've also tried the default regex and allowing extra characters before the dot(s)
```

This seems to work in some places and not in others:
* When running `dirvish`, it works in the middle buffer (current directory) but not the buffer for the parent directory.
* When running `dirvish-side`, it works in the top-level directory but not directories expanded in subtrees

(Note that I'm using emacs 29 and the latest on the dirvish `main` branch.)

I see that this appears to be working in the project's screenshots. What am I doing wrong?

**Comment by mikezackles** [comment]
I was able to work around this by using
```elisp
  (setq dired-listing-switches "-aBhl --group-directories-first --ignore=. --ignore=..")
```

I'm unsure if there are any hidden problems with this approach, but it seems to work everywhere.

**Comment by RedHatter** [comment]
You're looking for the `--almost-all` switch.

```elisp
(setq dired-listing-switches "-Bhl --group-directories-first --almost-all")
```

From `man ls`.

```
-A, --almost-all
        do not list implied . and ..
```

**Comment by Thaodan** [comment]
#268 should fix this problem by default without any interventions as . and .. are hidden by default using `dired-omit-files`.
